From 51cbd040814f8b08c05369c12ec73200923da4fe Mon Sep 17 00:00:00 2001
From: Biswapriyo Nath <nathbappai@gmail.com>
Date: Sun, 14 Sep 2025 12:00:41 +0200
Subject: [PATCH] lib: Fix build with libc++

src/parser.hh:1407:29: error: no member named 'max' in namespace 'std'
src/boxed.hh:39:30: error: no type named 'in_place_t' in namespace 'std'

Fixes: https://gitlab.gnome.org/GNOME/vte/-/merge_requests/8
---
 src/boxed.hh  | 1 +
 src/parser.hh | 1 +
 2 files changed, 2 insertions(+)

diff --git a/src/boxed.hh b/src/boxed.hh
index 4d4b07bcb..a526b5926 100644
--- a/src/boxed.hh
+++ b/src/boxed.hh
@@ -19,6 +19,7 @@
 // but we need this for non-enum/integral/floating types.
 
 #include <type_traits>
+#include <utility>
 
 namespace vte {
 
diff --git a/src/parser.hh b/src/parser.hh
index 071e506af..27c6d8fe4 100644
--- a/src/parser.hh
+++ b/src/parser.hh
@@ -18,6 +18,7 @@
 
 #pragma once
 
+#include <algorithm>
 #include <cstdint>
 #include <cstdio>
 #include <optional>
-- 
GitLab

